### DISCUSSION TITLE
Add BZip3 support

### DISCUSSION THREAD

**Discussion by sorairolake** [klauspost/compress#812] [Ideas/Features]
[BZip3](https://github.com/kspalaiologos/bzip3) is a compression algorithm developed in 2022. Some package managers support this[^1]. As far as I know, there is no Go implementation of this yet.

Would it be possible to add this algorithm to this library?

[^1]: https://repology.org/project/bzip3/versions

**Comment by klauspost** [comment]
I am generally not interested in making an LGPL package.

If the documentation was good enough for a clean room implementation I would consider it. But with the non-existing/extremely bare bones "specs", I would be forced to look at the code, which would pretty much force me to use LGPL (or compatible)

Either way, I am more likely to do improvements to zstd for the time being.

It *is* quite good, though.

**Reply by klauspost** [reply]
Adding a BWT to S2 blocks could be interesting.

Maybe I should check how much BWT+entropy coding and making offsets 3 bytes at most would improve compression.

Not sure how feasible it is given the slow decompression.
